commit 562089d2a1c216fbe49654da81b64a0f808e3b56
Author: John Spray <jspray@redhat.com>
Date:   Sat Feb 4 21:20:44 2017 +0000

    mgr: expose Objecter to PyModules
    
    Same motivation as exposing MonClient: so that modules
    can get at the command sending code.
    
    Signed-off-by: John Spray <john.spray@redhat.com>

diff --git a/src/mgr/PyModules.cc b/src/mgr/PyModules.cc
index b8c077cecee..0c28ef2e91f 100644
--- a/src/mgr/PyModules.cc
+++ b/src/mgr/PyModules.cc
@@ -61,8 +61,11 @@ namespace {
 #define dout_prefix *_dout << "mgr " << __func__ << " "
 
 PyModules::PyModules(DaemonStateIndex &ds, ClusterState &cs, MonClient &mc,
+                     Objecter &objecter_,
 		     Finisher &f)
-  : daemon_state(ds), cluster_state(cs), monc(mc), finisher(f)
+  : daemon_state(ds), cluster_state(cs), monc(mc),
+    objecter(objecter_),
+    finisher(f)
 {}
 
 // we can not have the default destructor in header, because ServeThread is
diff --git a/src/mgr/PyModules.h b/src/mgr/PyModules.h
index 6c6361a8125..0005dabb1b0 100644
--- a/src/mgr/PyModules.h
+++ b/src/mgr/PyModules.h
@@ -32,6 +32,7 @@ class PyModules
   DaemonStateIndex &daemon_state;
   ClusterState &cluster_state;
   MonClient &monc;
+  Objecter &objecter;
   Finisher &finisher;
 
   mutable Mutex lock{"PyModules"};
@@ -42,12 +43,14 @@ public:
   static constexpr auto config_prefix = "mgr.";
 
   PyModules(DaemonStateIndex &ds, ClusterState &cs, MonClient &mc,
+            Objecter &objecter_,
             Finisher &f);
 
   ~PyModules();
 
   // FIXME: wrap for send_command?
   MonClient &get_monc() {return monc;}
+  Objecter  &get_objecter() {return objecter;}
 
   PyObject *get_python(const std::string &what);
   PyObject *get_server_python(const std::string &hostname);
